docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,656 @@
+
+Usually, Java frameworks do their bootstrapping during the runtime before actually starting your application "Business oriented layer". During the bootstrap frameworks dynamically collect metadata by scanning the classpath to find configurations, entity definitions, dependency injection binding, etc. in order to instantiate proper objects through reflection. The main consequences are:
+
+* Delaying the readiness of your application: you need to wait couple of seconds before actually serve a business request.
+* Having a peak of resource consumption at bootstrap: In a constrained environment, you will need to size the needed resources based on your technical boostrap need rather than your actual business needs.
+
+Quarkus' philosophy is to prevent as much as possible slow and memory intensive dynamic code execution by shifting left these actions and eventually do them during the build time. A Quarkus extension is a Java piece of code acting as an adapter layer for your favorite libary or technology.
+
+== Description of a Quarkus extension
+
+A Quarkus extension consists of two parts:
+
+* Runtime module : Represent the capabilities the extension's developer exposes to the application's developer (an authentication filter, an enhanced data layer API, etc).
+* Deployment module: Used during the Augmentation build time, it describes how to "deploy" a library
+following the Quarkus philosophy. In other words, it applies all the Quarkus optimizations to your application before actually running it.
+
+At this point, you should have understood that most of the magic will happen at the Augmentation build time thanks to the deployment module.
+
+== Quarkus Application Bootstrap
+
+There are three distinct bootstrap phases of a Quarkus application.
+
+* Augmentation: During the build time, the Quarkus extensions will load and scan your application's bytecode (including the dependencies) and configuration. Once all the metadata collected, the extensions can pre-process the libraries bootstrap actions like your ORM, DI or REST controllers configurations. The result of the bootstrap is directly recorded into bytecode and will be part of your final application package.
+* Static Init: During the run time, Quarkus will execute first a static init method which contains some extensions actions/configurations. When you will do your native packaging, this static method will be pre-processed during the build time and the eventual generated objects will be serialized in the final native executable, so this code will not be executed in a native mode (Imagine you execute a Fibonacci function during this phase, the result of the computation will be directly recorded in the native executable). In a JVM mode, there is no much difference.
```

Comment:
Clarify the JVM effect. 
s/In a JVM mode, there is no much difference/When running the application in JVM, this static init phase is executed at the start of the application/